tests/e2e/apps_test.go

```diff
@@ -504,6 +504,26 @@ var _ = Describe("Apps", func() {
 			})
 		})
 
+		When("the app is restarted after binding to a user-provided service instance", func() {
+			var result map[string]interface{}
+
+			BeforeEach(func() {
+				var err error
+				serviceInstanceGUID := createServiceInstance(space1GUID, generateGUID("service-instance"))
+				createServiceBinding(appGUID, serviceInstanceGUID)
+				resp, err = certClient.R().SetResult(&result).Post("/v3/apps/" + appGUID + "/actions/restart")
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("sets the SERVICE_BINDING_ROOT env variable", func() {
```

Comment:
I thought about it and I am kind of on the fence, I can go either way. 
I felt like adding tests for checking whether correct credentials are mounted/unmounted under `$SERVICE_BINDING_ROOT` would feel like we are writing e2e to test the service binding controller itself.

Reply:
I'm also on the fence, but to play devil's advocate it would also be testing that we are associating Service Bindings with the correct workloads in a way that the Service Binding reconciler can understand. For example, I think this part of the code may be fragile:
https://github.com/cloudfoundry/korifi/blob/c06fb02679e12641e97f508e50aae68c256babfa/controllers/controllers/services/cfservicebinding_controller.go#L191

I think if that broke then that's also likely to break the `$SERVICE_BINDING_ROOT` env var injection, but I feel that assertion alone might be slightly weaker -- like I could imagine a world where the reconciler starts injecting `$SERVICE_BINDING_ROOT` into everything regardless of what is actually bound.

Reply:
okay, I ended up introducing a new `golang` app that returns the secrets mounted at `SERVICE_BINDING_ROOT` and its associated secret data as a json response. The e2e tests exercise both scenarios where credentials are mounted/unmounted

Example responses
```
$ curl -k https://golang.apps-127-0-0-1.nip.io/
go, world!

$ curl -k https://golang.apps-127-0-0-1.nip.io/servicebindingroot
/bindings

$ curl -k https://golang.apps-127-0-0-1.nip.io/servicebindings
{"2436d6e3-60d6-4f0c-b4eb-8f4026c57d47":{"password":"strong-pass","type":"user-provided","url":"sample.com","username":"bob"}}
```

Reply:
Awesome!